### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by MaksimBankou**

### CODE REVIEW

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package by.gsu.com.Interface;
+
+import by.gsu.com.WiFiZoneMetricsCalculator;
+
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java]: Please, add required comments here

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+public interface WiFiMetricsBuilder {
+    WiFiZoneMetricsCalculator build();
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java]: Please, add required comments here

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package by.gsu.com.Interface;
+
+import by.gsu.com.WiFiZoneMetricsCalculator;
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java:3]: Please, add required comments here

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+public interface WiFiMetricsFactory {
+    WiFiZoneMetricsCalculator getInstance();
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java]: Please, add required comments here

File: Wi-Fi Zone/src/main/java/by/gsu/com/Runner.java [hunk 5]
```diff
@@ -1,42 +1,17 @@
...
-
-            LOG.info("Value of calculate: " + metricsCalculator.calculate());
-            LOG.info("Value of free space loss: " + metricsCalculator.valuesFSL());
-            LOG.info("Value of the range of the Wi-Fi signal: " + metricsCalculator.length());
-        } catch (FileNotFoundException e) {
-            LOG.info("File not found");
-        } catch (IOException e) {
-            LOG.info("Download failed");
-        }
+        WiFiMetricsFactory wiFiMetricsFactory = new WiFiMetricsFilePropertiesFactoryImpl("src/main/java/resources/values.properties");
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Runner.java]: Please, move file path into separated constant

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
+package by.gsu.com;
+
+import by.gsu.com.Interface.WiFiMetricsFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+public class WiFiMetricsFilePropertiesFactoryImpl implements WiFiMetricsFactory {
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java:16]: Please, add required comments here

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            /**
+             * Data transfer
+             */
+            Double transmitterPower = Double.valueOf(properties.getProperty("transmitterPower"));
+            Double transmitAntennaGain = Double.valueOf(properties.getProperty("transmitAntennaGain"));
+            Double receiveAntennaGain = Double.valueOf(properties.getProperty("receiveAntennaGain"));
+            Double receiverSensitivity = Double.valueOf(properties.getProperty("receiverSensitivity"));
+            Double frequency = Double.valueOf(properties.getProperty("frequency"));
+            Double signalLoss1 = Double.valueOf(properties.getProperty("signalLoss1"));
+            Double signalLoss2 = Double.valueOf(properties.getProperty("signalLoss2"));
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: You use the same logic for all this variables, so create a method which get a property key and return it's value

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+            return new WiFiZoneMetricsCalculator(transmitterPower, transmitAntennaGain, receiveAntennaGain, receiverSensitivity, frequency, signalLoss1, signalLoss2);
+        } catch (IOException e) {
+            LOG.info("File not found");
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: Exception - is not just a simple information, like "file saved", you need to use higher logging level

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        return null;
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: When your method returns single instance it is better to use Optional in Java8, so refactor you interface and this method accordingly

File: Wi-Fi Zone/src/main/java/by/gsu/com/Runner.java [hunk 10]
```diff
@@ -1,42 +1,17 @@
...
+        WiFiMetricsBuilder wiFiMetricsBuilder = new WiFiMetricsFileWiFiMetricsBuilderImpl(wiFiMetricsFactory.getInstance());
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Runner.java]: wiFiMetricsFactory can be null here and you can get NPE. Read comment below about Optional and refactor this code

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    public String filePath;
+    /**
+     * Сlass call logger
+     */
+    private static Logger LOG = Logger.getLogger(WiFiZoneMetricsCalculator.class.getName());
+
+    /**
+     * filePath - The path to the file
+     * @param filePath
+     */
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: Refactor this javaDoc, please, it looks weird :)

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    /**
+     * Method reading values from a file
+     * @return null
+     */
+    public WiFiZoneMetricsCalculator getInstance() {
+        FileInputStream fis = null;
+
+        try {
+
+            fis = new FileInputStream(filePath);
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: Close this stream in a finally block

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+public class WiFiMetricsFilePropertiesFactoryImpl implements WiFiMetricsFactory {
+
+    /**
+     * Variable through which the file path is passed
+     */
+    public String filePath;
+    /**
+     * Сlass call logger
+     */
+    private static Logger LOG = Logger.getLogger(WiFiZoneMetricsCalculator.class.getName());
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: You shouldn't comment class fields, so you can remove it for all variables

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    private static Logger LOG = Logger.getLogger(WiFiZoneMetricsCalculator.class.getName());
+    /**
+     * Variable of the class WiFiZoneMetricsCalculator for calling methods
+     */
+    WiFiZoneMetricsCalculator wiFiZoneMetricsCalculator;
+
+    /**
+     * wiFiZoneMetricsCalculator - variable of the class WiFiZoneMetricsCalculator
+     * @param wiFiZoneMetricsCalculator
+     */
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Refactor this javaDoc, please, it looks weird :)

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    public WiFiMetricsFileWiFiMetricsBuilderImpl(WiFiZoneMetricsCalculator wiFiZoneMetricsCalculator) {
+        this.wiFiZoneMetricsCalculator = wiFiZoneMetricsCalculator;
+    }
+
+    /**
+     * Class for creating a file and recording results
+     * @return null
+     */
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Refactor this javaDoc, please, it looks weird :) Is it CLASS???

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+import by.gsu.com.Interface.WiFiMetricsBuilder;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+public class WiFiMetricsFileWiFiMetricsBuilderImpl implements WiFiMetricsBuilder {
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Please, add required comments here

Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Please, change it's name to WiFiMetricsFilePropertiesWiFiMetricsBuilderImpl cause you work with properties here

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+    /**
+     * Class for creating a file and recording results
+     * @return null
+     */
+    public WiFiZoneMetricsCalculator build(){
+        /**
+         * The path to the file
+         */
+        File file = new File("src/main/resources/valuesResults.properties");
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Move this file path into separated constant

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+        try {
+            /**
+             * File creation
+             */
+            BufferedWriter bw = new BufferedWriter(new FileWriter(file,true));
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Close this resource into final block

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+            /**
+             * Write to file
+             */
+            bw.write("Total system gain(Y) = " + wiFiZoneMetricsCalculator.calculate() + "\n" +
+                    "Free space loss(FSL) = " + wiFiZoneMetricsCalculator.valuesFSL() + "\n" +
+                    "The range of the Wi-Fi signal = " + wiFiZoneMetricsCalculator.length() + "\n");
+            bw.flush();
+            return wiFiZoneMetricsCalculator;
+        } catch (IOException e) {
+            LOG.info("Wrong path");
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: The same, it is not an info

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+             */
+            BufferedWriter bw = new BufferedWriter(new FileWriter(file,true));
+            /**
+             * Write to file
+             */
+            bw.write("Total system gain(Y) = " + wiFiZoneMetricsCalculator.calculate() + "\n" +
+                    "Free space loss(FSL) = " + wiFiZoneMetricsCalculator.valuesFSL() + "\n" +
+                    "The range of the Wi-Fi signal = " + wiFiZoneMetricsCalculator.length() + "\n");
+            bw.flush();
+            return wiFiZoneMetricsCalculator;
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Why do you try to return calculator? You change nothing in it. Refactor this code to return building state in boolean. Like  true - file created and everything is OK, and false if something went wrong

**Review by YauheniKhamiakou — CHANGES_REQUESTED** [review]

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java (hunk 1 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java]: Fix

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java (hunk 2 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java (hunk 3 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java:3]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java (hunk 4 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/Runner.java (hunk 5 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/Runner.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java (hunk 16 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java (hunk 17 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java (hunk 18 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java (hunk 19 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java (hunk 20 above)
Comment (MaksimBankou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFileWiFiMetricsBuilderImpl.java]: Fixed

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java [hunk 21]
```diff
@@ -0,0 +1,11 @@
+package by.gsu.com.Interface;
+/*
+ * This interface creates a method to create the file and records the values
+ * */
+import by.gsu.com.WiFiZoneMetricsCalculator;
+/*
+This method to create the file and records the values
+ */
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsBuilder.java:5]: WTF??? Comments for import???? and not for method???

File: .idea/vcs.xml [hunk 22]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (YauheniKhamiakou) [inline .idea/vcs.xml:6]: You shouldn't commit any more.

File: Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java [hunk 23]
```diff
@@ -0,0 +1,11 @@
+package by.gsu.com.Interface;
+/*
+ * This interface creates a method to reads the file
+ * */
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/Interface/WiFiMetricsFactory.java]: THE SAME SHIT AS ABOVE

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java (hunk 11 above)
Comment (YauheniKhamiakou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: NOT FIXED

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+     * @param filePath
+     */
+    public WiFiMetricsFilePropertiesFactoryImpl(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Method reading values from a file
+     * @return null
+     */
```
Comment (YauheniKhamiakou) [inline Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: BULLSHIT

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java (hunk 7 above)
Comment (YauheniKhamiakou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: NOT FIXED

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java (hunk 8 above)
Comment (YauheniKhamiakou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: NOT FIXED

File: Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java (hunk 9 above)
Comment (YauheniKhamiakou) [reply Wi-Fi Zone/src/main/java/by/gsu/com/WiFiMetricsFilePropertiesFactoryImpl.java]: NOT FIXED

**Review by YauheniKhamiakou — CHANGES_REQUESTED** [review]

**Pull request merged**